### PR TITLE
🌱 Bump Go to 1.20.6

### PR DESCRIPTION
**Pull request by chrischdi**
**What this PR does / why we need it**:

Bumps Go to the latest supported 1.20.6 to be on the same version as CAPI and k8s v1.27 / v1.28

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note
Bump Go to 1.20.6
```

### CODE REVIEW

**Comment by sbueringer** [comment]
Changes look okay to me. But CI doesn't agree :).

I wonder if we also have to bump the go version in ProwJobs before we can merge this PR

File: .github/workflows/golangci-lint.yaml [hunk 1]
```diff
@@ -12,11 +12,13 @@ jobs:
...
+      - name: Set up Go
+        uses: actions/setup-go@fac708d6674e30b6ba41289acaab6d4b75aa0753 # tag=v4.0.1
         with:
-          go-version: 1.19
+          go-version: ${{ steps.vars.outputs.go_version }}
       - name: golangci-lint
-        uses: golangci/golangci-lint-action@v3.6.0
-        with:
-          version: v1.52.2
+        uses: golangci/golangci-lint-action@639cd343e1d3b897ff35927a75193d57cfcba299 # tag=v3.6.0
```
Comment (sbueringer) [inline .github/workflows/golangci-lint.yaml:24]: I think we should still pin the golangci-lint version

Comment (sbueringer) [reply .github/workflows/golangci-lint.yaml:24]: It's probably already part of what you're planning to do. Let's also get rid of the shell script to install golangci-lint and align to core CAPI

Comment (chrischdi) [reply .github/workflows/golangci-lint.yaml]: Let me take bumping the linter out to a separate PR :-)

Comment (sbueringer) [reply .github/workflows/golangci-lint.yaml]: Absolutely!

**Pull request merged**